packages/flutter_test/lib/src/accessibility.dart

```diff
@@ -196,131 +229,180 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+    final List<Element> elements =
+        find.text(text).hitTestable().evaluate().toList();
+    late final Rect paintBounds;
+
+    if (elements.length == 1) {
+      final Element element = elements.single;
+      final RenderObject? renderBox = element.renderObject;
+      if (renderBox is! RenderBox) {
+        throw StateError('Unexpected renderObject type: $renderBox');
       }
-      final double contrastRatio = report.contrastRatio();
-      const double delta = -0.01;
-      double targetContrastRatio;
-      if ((isBold && (fontSize ?? _kDefaultFontSize) > kBoldTextMinimumSize) || (fontSize ?? _kDefaultFontSize) > kLargeTextMinimumSize) {
-        targetContrastRatio = kMinimumRatioLargeText;
+
+      const Offset offset = Offset(4.0, 4.0);
+      paintBounds = Rect.fromPoints(
+        renderBox.localToGlobal(renderBox.paintBounds.topLeft - offset),
+        renderBox.localToGlobal(
+          renderBox.paintBounds.bottomRight + offset,
+        ),
+      );
+      final Widget widget = element.widget;
+      final DefaultTextStyle defaultTextStyle = DefaultTextStyle.of(element);
+      if (widget is Text) {
+        final TextStyle? style = widget.style;
+        final TextStyle effectiveTextStyle = style == null || style.inherit
+            ? defaultTextStyle.style.merge(widget.style)
+            : style;
+        isBold = effectiveTextStyle.fontWeight == FontWeight.bold;
+        fontSize = effectiveTextStyle.fontSize;
+        textColor = effectiveTextStyle.color;
+      } else if (widget is EditableText) {
+        isBold = widget.style.fontWeight == FontWeight.bold;
+        fontSize = widget.style.fontSize;
+        textColor = widget.style.color;
       } else {
-        targetContrastRatio = kMinimumRatioNormalText;
-      }
-      if (contrastRatio - targetContrastRatio >= delta) {
-        return result + const Evaluation.pass();
+        throw StateError('Unexpected widget type: ${widget.runtimeType}');
       }
-      return result + Evaluation.fail(
-        '$node:\nExpected contrast ratio of at least '
-        '$targetContrastRatio but found ${contrastRatio.toStringAsFixed(2)} for a font size of $fontSize. '
-        'The computed light color was: ${report.lightColor}, '
-        'The computed dark color was: ${report.darkColor}\n'
-        'See also: https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html'
+    } else if (elements.length > 1) {
+      return Evaluation.fail(
+        'Multiple nodes with the same label: ${data.label}\n',
       );
+    } else {
+      // If we can't find the text node then assume the label does not
+      // correspond to actual text.
+      return result;
+    }
+
+    if (isNodeOffScreen(paintBounds, tester.binding.window)) {
+      return result;
+    }
+
+    final Map<Color, int> colorHistogram =
+        _colorsWithinRect(byteData, paintBounds, image.width, image.height);
+
+    // Node was too far off screen.
```

Comment:
in which cases does this happen when `isNodeOffScreen` is false?

Reply:
If element is outside screen but still in semantic tree.

Reply:
consider renaming method / editing comment to reflect this

Reply:
Updated method comment.